test/resources/audioWorkletStream.html

```diff
@@ -0,0 +1,75 @@
+<!DOCTYPE html>
+
+<html>
+  <head> </head>
+
+  <body>
+    <h1>AudioWorklet Transform Stream</h1>
+    <script type="worklet">
+      class AudioWorkletStream extends AudioWorkletProcessor {
+        constructor(options) {
+          super();
+          console.log(this);
+          this.port.onmessage = e => {
+            this.writable = e.data;
+            this.writer = this.writable.getWriter();
+            this.port.postMessage('stream')
+          };
+        }
+        process(inputs, outputs) {
+          // convert to Int16Array, UintArray here
+          const channels = inputs.flat().reduce((f32, channel, index) => (f32.set(channel, !index ? index : 128), f32), new Float32Array(256));
+          this.writer.write(channels);
+          return true;
+        }
+      };
+      registerProcessor(
+        'audio-worklet-stream',
+        AudioWorkletStream
+      );
+    </script>
+    <script>
+      document.querySelector('h1').onclick = async () => {
+        const mediaStream = await navigator.mediaDevices.getUserMedia({
+          audio: true,
+        });
+        console.log(mediaStream);
+        const worklet = URL.createObjectURL(
+          new Blob(
+            [document.querySelector('script[type=worklet]').textContent],
+            {
+              type: 'text/javascript',
+            }
+          )
+        );
```

Comment:
Nice! I was hoping there would be a way to do this without creating separate files!